compiler/rustc_hir_analysis/src/check/method/suggest.rs

```diff
@@ -1201,6 +1204,81 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         false
     }
 
+    fn suggest_range_for_iter(
```

Comment:
this name doesn't feel completely right anymore, maybe
```suggestion
    fn suggest_wrapping_range_with_parens(
```

Reply:
also please add a short comment explaining the purpose of this function, even just something like

```rust
/// When encountering `0..1.map(|i| i + 1)` suggest `(0..1).map(|i| i + 1)`.
```